src/bin/common/solving_loop.ml

```diff
@@ -175,7 +186,7 @@ let main () =
       exit 142
   in
 
-  let solver_ctx_key: solver_ctx State.key =
+  let solver_ctx_key: SAT.t solver_ctx State.key =
```

Comment:
Instead of parameterizing the `solver_ctx` by a `SAT.t`, we should create two different keys for the `solver_ctx` and the `sat_ctx`.

```ocaml
let sat_state_key : SAT.t State.key = State.create_key ~pipe:"" "sat_state" in
```